### DISCUSSION TITLE
Opening PHPstorm from Ray (Windows)

### DISCUSSION THREAD

**Discussion by colinmackinlay** [spatie/laravel-ray#65] [Q&A] [answered]
When clicking on the link in the Ray window:
![Capture](https://user-images.githubusercontent.com/7833362/104247599-04db3300-5460-11eb-88ab-19bda55728df.PNG)
Windows doesn't recognise that PHPstorm is to be called and gives an `you'll need a new app to open this phpstorm link` message.

The only option available is to browse the AppStore but that doesn't find PHPstorm. Has anyone got that working on Windows yet? I'm guessing there must be a way to set an association but I certainly can't find it! Thanks :)

**Comment by amacsmith** [comment]
I am having that same issue. However I was not even able to get the host machine path to work. It all points to my homestead path.

**Comment by amacsmith** [comment, marked as answer]
I actually figured out a way to do this @colinmackinlay. I downloaded https://github.com/aik099/PhpStormProtocol and followed the directions. Be sure to update the run_editor.js with the current version of php storm like show below and everything should work!
![image](https://user-images.githubusercontent.com/25642759/104360597-5545a580-54df-11eb-9a27-66daa98dab10.png)

**Reply by freekmurze** [reply]
@amacsmith Nice! Can you PR these instructions to our docs?

**Reply by amacsmith** [reply]
yes

**Reply by colinmackinlay** [reply]
Nice job @amacsmith 
When you do the PR, perhaps worth adding a note that if you use Toolbox rather than a direct Install of PHPstorm then you also need to change the first flag to `true`.

My run_editor.js now starts like this:
```
var settings = {
    // flag to active Jetbrain Toolbox configuration
    toolBoxActive: true,

    // Set to 'true' (without quotes) if run on Windows 64bit. Set to 'false' (without quotes) otherwise.
    x64: true,

    // Set to folder name, where PhpStorm was installed to (e.g. 'PhpStorm')
    folder_name: 'PhpStorm 2020.3',

    // Set to window title (only text after dash sign), that you see, when switching to running PhpStorm instance
    window_title: 'PhpStorm 2020.3',

    // In case your file is mapped via a network share and paths do not match.
    // eg. /var/www will can replaced with Y:/
    projects_basepath: '',
    projects_path_alias: ''
};
```
Thanks again

**Reply by amacsmith** [reply]
I have created the PR [#126](https://github.com/spatie/ray/pull/126)

**Comment by colinmackinlay** [comment]
Hey @amacsmith, a tiny wrinkle that remains seems to be that the link causes PHPstorm to open a new "instance" rather than open the file in the currently open project. Is that what happens for you?

I suspect it's a PHPstorm thing and nothing to do with Ray but it would be interesting to know if that's the behaviour on Mac/Linux too anyone? It's not the end of the world but if you then try to go exploring further when debugging using say Search Anywhere or Find Usage then PHPstorm can't respond as it has no context.

**Reply by amacsmith** [reply]
I'm not at my pc. However, I definitely would have noticed that. So I believe that is a phpstorm thing. I will expand or update in a minute when I get a chance.

**Reply by amacsmith** [reply]
If you have that project open and the workspace set. It shouldn't open a new window.  However, check your phpstorm settings here File | Settings | Appearance & Behavior | System Settings. Also here is what I am referencing [phpstorm open new window](https://intellij-support.jetbrains.com/hc/en-us/community/posts/206372659-How-do-I-open-a-new-window).  Let me know your results!

**Reply by colinmackinlay** [reply]
Thanks - makes no difference unfortunately. I've set and reset that option, behaviour is unaffected.

It doesn't open a full "instance" just a single editing window with limited functionality ..... hmmm!
![Capture](https://user-images.githubusercontent.com/7833362/104388853-fbba9680-5531-11eb-9171-bd0f038b3352.PNG)

**Reply by amacsmith** [reply]
Yeah just had a moment to check it out on my pc and it does not open a new window. It opens it in the workspace instance that the file resides in. What version is your PhpStorm on? Try my settings below and maybe you may get a different effect. Make sure to close and reopen to ensure it resets. Also @freekmurze may have some insight into that setting as well.

![image](https://user-images.githubusercontent.com/25642759/104390837-9ef8b500-550c-11eb-85c5-d091e450e722.png)

**Reply by colinmackinlay** [reply]
Thanks - those were my settings and a reboot of machine didn't help. But...

Ray updated today and there was one more update of PHPstorm available to 2020.3.1. Between them that fixed it and now it works fine including launching and opening the correct project even if PHPstorm isn't open. Thanks for your help.

**Reply by amacsmith** [reply]
No problem!
